internal/handlers/pg/pgdb/query.go

```diff
@@ -246,20 +246,38 @@ func prepareWhereClause(sqlFilters *types.Document) (string, []any) {
 	var p Placeholder
 
 	for k, v := range sqlFilters.Map() {
-		switch k {
-		case "_id":
-			switch v := v.(type) {
-			case string:
-				filters = append(filters, fmt.Sprintf(`((_jsonb->'_id')::jsonb = %s)`, p.Next()))
-				args = append(args, string(must.NotFail(pjson.MarshalSingleValue(v))))
+		// don't pushdown query operators and dot notation yet
+		if len(k) != 0 && (k[0] == '$' || types.NewPathFromString(k).Len() > 1) {
+			continue
+		}
 
-			case types.ObjectID:
+		// don't iterate through array for _id keys to simplify the query
+		if k == "_id" {
+			switch v := v.(type) {
+			case types.ObjectID, string:
 				filters = append(filters, fmt.Sprintf(`((_jsonb->'_id')::jsonb = %s)`, p.Next()))
 				args = append(args, string(must.NotFail(pjson.MarshalSingleValue(v))))
 			}
-		default:
+
 			continue
 		}
+
+		switch v := v.(type) {
+		case string, int32, int64, types.ObjectID, float32, float64:
```

Comment:
Let's order them to align with our convention and we don't have float32.
https://github.com/FerretDB/FerretDB/blob/main/CONTRIBUTING.md#code-style-and-conventions.

I suggest listing all data types and `continue` on the types we don't support yet. :hugs: That way, it's easier to see what's not supported

Reply:
Whoops! I leave too many thing while I'm debugging :smile: nice catch!